pkg/fqdn/service/service.go

```diff
@@ -102,17 +123,112 @@ type PolicyUpdater interface {
 
 // StreamPolicyState is a bidirectional streaming RPC to subscribe to DNS policies
 // SDP calls this method to subscribe to DNS policies
-// For each stream, we start a goroutine to receive the DNS policies ACKs
+// For each stream, we start a goroutine to receive the DNS policies ACKs for that particular client.
 // The flow of the method is as follows:
-// 1. Add the stream to the map( called by the client i.e SDP)
-// 2. Start a goroutine to receive the DNS policies ACKs for that particular client.
-// 3. Send the current state of the DNS rules to the client (We store the current state fo DNS rules during the endpoint regeneration see UpdatePolicyRulesLocked)
-// 4. Wait for the context to be done
-// Note: this method is left empty on purpose and will be update with the actual implementation in the future PRs for the standalone DNS proxy
+//  1. Add the stream to the local map(called by the client i.e SDP)
+//  2. Start a goroutine to receive the DNS policies ACKs for that particular client.
+//  3. Send the current state of the DNS rules to the client (We store the current state of
+//     DNS rules during the endpoint regeneration see UpdatePolicyRules)
+//  4. Wait for the context to be done and return the error if any
+//  5. Delete the stream from the local map
 func (s *FQDNDataServer) StreamPolicyState(stream pb.FQDNData_StreamPolicyStateServer) error {
-	// This is a temporary implementation to send the current state of the DNS rules to the client and used for testing
-	stream.Send(&pb.PolicyState{RequestId: "test"})
-	return nil
+	streamCtx, cancel := context.WithCancel(stream.Context())
+	s.streams.Store(stream, cancel)
+
+	ackRecvStream := make(chan struct{}, 1)
+	errChan := make(chan error)
+	// Start a goroutine to receive the DNS policies ACKs
```

Comment:
I'm not sure this needs to be a separate goroutine. I believe we can just do
```
stream.Send(...)
stream.Recv(...)
```
without any other synchronization points.

Reply:
The goroutine to receive ack is per stream (i.e per connected standalone dns proxy) so that goroutine also listens for any future updates to the DNS policy rules being sent to SDP.